packages/ens-utils/src/ethregistrar.ts

```diff
@@ -155,3 +179,284 @@ export function buildNFTRefFromENSName(
+  in the Ui. The reason we are separating this text in different fields is because:
+
+  1. We want to be able to display different texts depending on wether the price of
+  the domain is a premium price or not. In each one of these cases, the text displayed
+  is different.
+  2. Since the design for this data displaying is differently defined for the price field
+  and the descriptive text, we separate it so we can render these two fields separately in the
+  HTML that will be created inside the component. e.g. the price field is bold and the descriptive
+  text is not. Please refer to this Figma artboard for more details: https:/*www.figma.com/file/lZ8HZaBcfx1xfrgx7WOsB0/Namehash?type=design&node-id=12959-119258&mode=design&t=laEDaXW0rg9nIVn7-0
+*/
+export interface PriceDescription {
+  /* descriptiveTextBeginning references the text that is displayed before the price */
+  descriptiveTextBeginning: string;
+  /* pricePerYear is a string that represents: Price + "/ year" (e.g. "$5.99 / year") */
+  pricePerYearDescription: string;
+  /* descriptiveTextBeginning references the text that is displayed after the price */
+  descriptiveTextEnd: string;
+}
+
+export const getPriceDescription = (
+  registration: Registration,
+  parsedName: DomainName,
+): PriceDescription | null => {
+  const isExpired =
+    registration.primaryStatus === PrimaryRegistrationStatus.Expired;
+  const wasRecentlyReleased =
+    registration.secondaryStatus ===
+    SecondaryRegistrationStatus.RecentlyReleased;
+  const isRegistered =
+    registration.primaryStatus === PrimaryRegistrationStatus.Active;
+
+  if (!(isExpired && wasRecentlyReleased) && isRegistered) return null;
+  const domainBasePrice = AvailableNameTimelessPriceUSD(parsedName);
+
+  if (!domainBasePrice) return null;
+  else {
+    const domainPrice = formattedPrice({
+      price: domainBasePrice,
+      withPrefix: true,
+    });
+    const pricePerYearDescription = `${domainPrice} / year`;
+
+    const premiumEndsIn = premiumPeriodEndsIn(registration)?.relativeTimestamp;
+
+    if (premiumEndsIn) {
+      const premiumEndMessage = premiumEndsIn
+        ? ` Temporary premium ends ${premiumEndsIn}.`
+        : null;
+      const basePriceMessage = domainBasePrice
+        ? " Discounts continuously until dropping to "
+        : null;
+
+      return {
+        pricePerYearDescription,
+        descriptiveTextBeginning:
+          "Recently released." + premiumEndMessage + basePriceMessage,
+        descriptiveTextEnd: ".",
+      };
+    } else {
+      const domainLabelLength = parsedName.labelName.length;
+
+      return domainLabelLength <
+        DOMAIN_HAS_SPECIAL_PRICE_IF_LENGTH_EQUAL_OR_LESS_THAN
+        ? {
+            pricePerYearDescription,
+            descriptiveTextBeginning: `${domainLabelLength}-character names are `,
+            descriptiveTextEnd: " to register.",
+          }
+        : null;
+    }
+  }
+};
+
+export const nameCurrentTemporaryPremium = (
```

Comment:
```suggestion
export const registrationCurrentTemporaryPremium = (
```

Suggesting this as the param we pass in is a registration, not a name.